rinex-cli/src/positioning/interp/orbit.rs

```diff
@@ -0,0 +1,398 @@
+use crate::cli::Context;
+use gnss_rtk::prelude::{
+    AprioriPosition, Epoch, InterpolationResult as RTKInterpolationResult, SV,
+};
+use rinex::navigation::Ephemeris;
+use std::collections::HashMap;
+
+use super::Buffer as BufferTrait;
+
+struct Buffer {
+    order: usize,
+    inner: Vec<(Epoch, (f64, f64, f64))>,
+}
+
+impl BufferTrait<(f64, f64, f64)> for Buffer {
+    fn malloc(order: usize) -> Self {
+        if order % 2 == 0 {
+            panic!("only odd interpolation orders currently supported");
+        }
+        Self {
+            order,
+            inner: Vec::with_capacity(order + 1),
+        }
+    }
+    fn len(&self) -> usize {
+        self.inner.len()
+    }
+    /// Fill internal Buffer, which does not tolerate data gaps
+    fn at_capacity(&self) -> bool {
+        self.inner.len() == self.order + 1
+    }
+    fn clear(&mut self) {
+        self.inner.clear();
+    }
+    fn get(&self, idx: usize) -> Option<&(Epoch, (f64, f64, f64))> {
+        self.inner.get(idx)
+    }
+    fn latest(&self) -> Option<&(Epoch, (f64, f64, f64))> {
+        self.inner.last()
+    }
+    fn push(&mut self, x_j: (Epoch, (f64, f64, f64))) {
+        if self.len() == self.order + 1 {
+            let mut index = 0;
```

Comment:
Is this `self.inner.remove(0)`?

Reply:
By the way, this is going to be the least of your problems, but you can use a ring buffer to store the samples, i.e. keep track of the number of samples and index of the last one.

Reply:
I simplified the logic a bit, removed the trait and made sure it still works.
This behaves like a FIFO of a strict depth